packages/kas/src/__tests__/units_test.js

```diff
@@ -0,0 +1,189 @@
+        expected; // ?
+        return Math.round(answer.eval()) == Math.round(expected.eval())
+            ? {pass: true}
+            : {
+                  pass: false,
+                  message: () =>
+                      `${original} = ["${expected.print()}"] ${newUnit}`,
+              };
+    },
+    toHaveTheSameForm([x, y]: [string, string], msg: string) {
+        const equal = KAS.compare(
+            KAS.unitParse(x).unit,
+            KAS.unitParse(y).unit,
+        ).equal;
+
+        return equal ? {pass: true} : {pass: false, message: () => msg};
+    },
+    toHaveMagnitude(input: string, expected: number) {
+        const parsed = KAS.unitParse(input).coefficient;
+
+        return +parsed === expected
+            ? {pass: true}
+            : {
+                  pass: false,
+                  message: () => `magnitude of ${input} is not ${expected}`,
+              };
+    },
+});
+
+describe("units", () => {
+    test.only("simplify expressions with units", () => {
+        expect([
+            new KAS.Mul(new KAS.Rational(1, 100), new KAS.Unit("cup")),
+            new KAS.Mul(new KAS.Rational(1, 400), new KAS.Unit("qt")),
+        ]).toHaveEqualUnits("1/100 cup = 1 / 400 quart");
+
+        expect(["10 g", "0.01 kg"]).toParseUnitsAsEqual("10 g = 1 / 100 kg");
+
+        expect(["9.8 kg m / s^2", "9.8 N"]).toParseUnitsAsEqual(
+            "9.8 kg m / s^2 = 9.8 N",
+        );
+
+        expect(["9.8 m / s^2", "9.8 N / kg"]).toParseUnitsAsEqual(
+            "9.8 m / s^2 = 9.8 N / kg",
+        );
+
+        expect([
+            new KAS.Mul(
+                new KAS.Float(9.8),
+                new KAS.Unit("m"),
+                new KAS.Pow(new KAS.Unit("s"), new KAS.Int(-2)),
+            ),
+            new KAS.Mul(
+                new KAS.Float(9.8),
+                new KAS.Pow(new KAS.Unit("s"), new KAS.Int(-2)),
+                new KAS.Unit("m"),
+            ),
+        ]).toHaveEqualUnits("9.8 m / s^2 = 9.8 s^-2 m");
+
+        expect([
+            new KAS.Mul(
+                new KAS.Int(50),
+                new KAS.Unit("m"),
+                new KAS.Pow(new KAS.Unit("m"), new KAS.Int(-1)),
+            ),
+            new KAS.Int(50),
+        ]).toHaveEqualUnits("50 m / m = 50");
+
+        // There's a long chain of conversions before this is fully simplified.
+        // tsp -> tbsp -> cup -> gal -> L -> m^3
+        //
+        //         1 tbsp    1 cup     1 gal   3.785 L
+        // 1 tsp * ------ * ------- * ------ * -------
+        //         3 tsp    16 tbsp   16 cup     gal
+        expect([
+            new KAS.Unit("tsp"),
+            new KAS.Mul(
+                new KAS.Rational(1, 3),
+                new KAS.Rational(1, 16),
+                new KAS.Rational(1, 16),
+                new KAS.Float(3.785),
+                new KAS.Unit("L"),
+            ),
+        ]).toHaveEqualUnits("tsp reduces");
+
+        expect(
+            !KAS.compare(
```

Comment:
Have the negation here really confuses the assertion on 139. Might be worth removing the `!` and switching 139 to `.toBeFalse()`?